File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int sum = 0;
        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i) == 'I') {
                sum = sum + 1;                
            }
            else if(s.charAt(i) == 'V') {
                sum = sum + 5;
                if(i-1 >= 0 && s.charAt(i-1) == 'I') {
                    sum = sum - 2;
                }
            }
            else if(s.charAt(i) == 'X') {
                sum = sum + 10;
                if(i-1 >= 0 && s.charAt(i-1) == 'I') {
                    sum = sum - 2;
                }
            }
            else if(s.charAt(i) == 'L') {
                sum = sum + 50;
                if(i-1 >= 0 && s.charAt(i-1) == 'X') {
                    sum = sum - 20;
                }
            }
            else if(s.charAt(i) == 'C') {
                sum = sum + 100;
                if(i-1 >= 0 && s.charAt(i-1) == 'X') {
                    sum = sum - 20;
                }
            }
            else if(s.charAt(i) == 'D') {
                sum = sum + 500;
                if(i-1 >= 0 && s.charAt(i-1) == 'C') {
                    sum = sum - 200;
                }
            }
            else if(s.charAt(i) == 'M') {
                sum = sum + 1000;
                if(i-1 >= 0 && s.charAt(i-1) == 'C') {
                    sum = sum - 200;
                }
            }
        }
        return sum;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        Sum = 0

        for i in range(len(s)):
            if s[i] == 'I':
                Sum = Sum + 1

            elif s[i] == 'V':
                Sum = Sum + 5
                if i > 0 and s[i-1] == 'I':
                    Sum = Sum - 2

            elif s[i] == 'X':
                Sum = Sum + 10
                if i > 0 and s[i-1] == 'I':
                    Sum = Sum - 2

            elif s[i] == 'L':
                Sum = Sum + 50
                if i > 0 and s[i-1] == 'X':
                    Sum = Sum - 20

            elif s[i] == 'C':
                Sum = Sum + 100
                if i > 0 and s[i-1] == 'X':
                    Sum = Sum - 20

            elif s[i] == 'D':
                Sum = Sum + 500
                if i > 0 and s[i-1] == 'C':
                    Sum = Sum - 200

            elif s[i] == 'M':
                Sum = Sum + 1000
                if i > 0 and s[i-1] == 'C':
                    Sum = Sum - 200

        return Sum